sdks/java/io/hadoop/input-format/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIOIT.java

```diff
@@ -0,0 +1,133 @@
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.postgresql.ds.PGSimpleDataSource;
+
+/**
+ * IOIT for {@link org.apache.beam.sdk.io.hadoop.inputformat.HadoopInputFormatIO}.
+ */
+public class HadoopInputFormatIOIT {
+
+  private static PGSimpleDataSource dataSource;
+  private static Long numberOfRows;
+  private static String tableName;
+  private static SerializableConfiguration hadoopConfiguration;
+
+  @Rule
+  public TestPipeline writePipeline = TestPipeline.create();
+
+  @Rule
+  public TestPipeline readPipeline = TestPipeline.create();
+
+  @BeforeClass
+  public static void setUp() throws SQLException {
+    PipelineOptionsFactory.register(IOTestPipelineOptions.class);
+    IOTestPipelineOptions options = TestPipeline.testingPipelineOptions()
+        .as(IOTestPipelineOptions.class);
+
+    dataSource = DatabaseTestHelper.getPostgresDataSource(options);
+    numberOfRows = options.getNumberOfRecords();
+    tableName = DatabaseTestHelper.getTestTableName("HadoopInputFormatIOIT");
+
+    DatabaseTestHelper.createDataTable(dataSource, tableName);
+    setupHadoopConfiguration(options);
+  }
+
+  private static void setupHadoopConfiguration(IOTestPipelineOptions options) {
+    Configuration conf = new Configuration();
+    DBConfiguration.configureDB(
+        conf,
+        "org.postgresql.Driver",
+        DatabaseTestHelper.getPostgresDBUrl(options),
+        options.getPostgresUsername(),
+        options.getPostgresPassword()
+    );
+    conf.set(DBConfiguration.INPUT_TABLE_NAME_PROPERTY, tableName);
+    conf.setStrings(DBConfiguration.INPUT_FIELD_NAMES_PROPERTY, "id", "name");
+    conf.setClass(DBConfiguration.INPUT_CLASS_PROPERTY, TestRowDBWritable.class, DBWritable.class);
+
+    conf.setClass("key.class", LongWritable.class, Object.class);
+    conf.setClass("value.class", TestRowDBWritable.class, Object.class);
+    conf.setClass("mapreduce.job.inputformat.class", DBInputFormat.class, InputFormat.class);
+
+    hadoopConfiguration = new SerializableConfiguration(conf);
+  }
+
+  @AfterClass
+  public static void tearDown() throws SQLException {
+    DatabaseTestHelper.cleanUpDataTable(dataSource, tableName);
+  }
+
+  @Test
+  public void writeThenReadUsingDBInputFormat() {
+    writePipeline.apply("Generate sequence", GenerateSequence.from(0).to(numberOfRows))
+        .apply("Produce db rows", ParDo.of(new DeterministicallyConstructTestRowFn()))
+        .apply(JdbcIO.<TestRow>write()
+            .withDataSourceConfiguration(JdbcIO.DataSourceConfiguration.create(dataSource))
+            .withStatement(String.format("insert into %s values(?, ?)", tableName))
+            .withPreparedStatementSetter(new PrepareStatementFromTestRow()));
+
+
+    writePipeline.run().waitUntilFinish();
+
+    PCollection<String> consolidatedHashcode = readPipeline
+        .apply("Read using DBInputFormat", HadoopInputFormatIO
```

Comment:
<!--new_thread; commit:138b0e00d1ea86cd27700083ee657e8c5abae4a6; resolved:0-->
Prob. rename to "Read using HadoopInputFormat" since that explains better what we are testing here.